R/add_module.R

```diff
@@ -0,0 +1,323 @@
+#' Create a module
+#' 
+#' This function creates a `{tm}` module class inside the `R/` folder.
+#' 
+#' @param name The class name of the module. 
+#' @param pkg Path to the root of the package. Default is `getwd()`.
+#' @param inherit Parent module class. Default is TidyModule.
+#' @param open Should the file be opened?
+#' @param dir_create Creates the directory if it doesn't exist, default is `TRUE`.
+#' @param export Logical. Should the module be exported? Default is `FALSE`.
+#' @note As a convention, this function will automatically capitalize the first character of the `name` argument.
+#' 
+#' @importFrom cli cat_bullet
+#' @importFrom utils file.edit
+#' @importFrom fs path_abs path file_create
+#' @importFrom snippr snippets_read
+#' 
+#' @export
+add_module <- function(
+  name,
+  inherit = "TidyModule",
+  pkg = getwd(), 
```

Comment:
Does this function support only R package structure only? what about a regular shiny app? Probably need to document in the vignette on the good practice of organizing {tm} class files

Reply:
Here is my proposition (which is actually implemented in the function)
- the default path is always current wd => option `path`
- if R sub-dir exists, add it
- class name always starts with capital letter
- there is a default prefix added to filename "tm_" => option `prefix`

Reply:
The function supports both !